docs/concepts/core-workflow-concepts.md

```diff
@@ -0,0 +1,129 @@
+# Core Concepts
+
+## Introduction
+
+This document explains the core concepts that are needed to understand the workflow engine.
+
+## Workflow
+
+A workflow is the definition of a process. It is made up of states and transitions between those states. The following diagram shows a simple workflow representing the lifetime of an item in a catalogue:
+
+```
+                  Create
+                     |
+                     | 
+                     |
+             +-------v-------+
+             |               |
+             | Waiting for   |
+      +------+ documentation |
+      |      |               |
+      |      +-------+-------+
+      |              |
+Edit                 | Publish
+(incomplete)         |
+      |      +-------v-------+               +---------------+
+      |      |               |               |               |
+      +------+   Published   |  Deprecate    |  Deprecated   |
+             |               +--------------->    (end)      |
+      +------>               |               |               |
+      |      +--+---------+--+               +---------------+
+      |         |         |
+ Edit +---------+         | Delete
+ (complete)               |
+             +------------v--+
+             |               |
+             |    Deleted    |
+             |     (end)     |
+             |               |
+             +---------------+
+
+```
+
+This workflow contains four states and 5 transitions. The states are represented by boxes, and the transitions by the lines between them.
+
+As you can see from the diagram, the workflow starts with the "Waiting for documentation" state and finishes when either the "Deprecated" or "Deleted" states are reached. Transitions allow the item to cycle between the "Published" and "Waiting for documentation" states.
+
+## Workflow Instance
+
+A workflow instance is a representation of a specific item's progress through the workflow. The workflow above represents the lifetime of an item in a catalogue; each of those items will be associated with a workflow instance representing its state at any point.
+
+A workflow instance contains the id and version of its parent workflow, the current state of the item in the workflow, and any other useful data that's associated with that instance.
+
+Instances also have a Status. The main statuses are:
+- Initializing, which indicates that a workflow instance has been created but is not yet ready to receive triggers 
+- Waiting, which indicates that the instance is ready to receive triggers
+- Complete, which indicates that the workflow instance has entered a state that has no transitions
```

Comment:
True. I didn't explicitly call that out because in our model, transitions always belong to the state they are transitioning from - i.e. it can only ever have exit transitions.

If I'd explicitly called it out in that way, then to me it becomes more confusing - if a state has a transition that leads back to itself, would that still be an "exit transition"? (Yes, it would - but I think it would be easy to make the mistake.)